controllers/store.js

```diff
@@ -1,34 +1,150 @@
 
     //TODO: access the books loaned for this user and render loaned books page
+    BookCopy.find((err, bookCopies) => {
+        if (err) throw err;
+        Book.find((err, books) => {
+            res.render('loaned_books', {
+                books: bookCopies.filter(bookCopy => req.user.loaned_books.includes(bookCopy.id))
+                    .map(bookCopy => ({
+                        book: books.find(book => (book.id == bookCopy.book)),
+                        id: bookCopy.id,
+                        borrow_date: `${bookCopy.borrow_date.getDate()}/${bookCopy.borrow_date.getMonth() + 1}/${bookCopy.borrow_date.getFullYear()}`,
+                    })),
+                title: `Loaned Books | ${req.user.username}`,
+                success_message: req.query.ret == 'success' ? "Book returned successfully" : undefined,
+                error_message: req.query.ret == 'failure' ? "Book could not be returned" : undefined,
+            })
+        })
+    })
+
+
 }
 
 var issueBook = (req, res) => {
-    
     // TODO: Extract necessary book details from request
     // return with appropriate status
     // Optionally redirect to page or display on same
+    try {
+        BookCopy.find({ book: req.body.bid, status: true }, (err, bookCopies) => {
+            if (bookCopies.length == 0) {
+                res.redirect(`/book/${req.body.bid}/?issue=failure`);
+            } else {
+                bookCopies[0].status = false;
+                bookCopies[0].borrow_date = Date.now();
+                bookCopies[0].borrower = req.user.id;
+                bookCopies[0].save().then((_) => {
+                    req.user.loaned_books.push(bookCopies[0].id);
+                    req.user.save().then((_) => {
+                        res.redirect(`/book/${req.body.bid}/?issue=success`);
+                    })
+                });
+            }
+        });
+    } catch {
+        res.redirect(`/book/${req.body.bid}/?issue=failure`);
+
+    }
+}
+
+var returnBook = (req, res) => {
+    try {
+        BookCopy.findById(req.body.bcid, (err, bookCopy) => {
+            if (err) throw err;
+            if (bookCopy.status) {
+                res.send('Book was already Returned');
+            } else {
+                bookCopy.status = true;
+                bookCopy.borrow_date = null;
+                bookCopy.borrower = null;
+                req.user.loaned_books.splice(req.user.loaned_books.indexOf(req.body.bcid), 1)
+                bookCopy.save().then((_) => {
+                    req.user.save().then((_) => {
+                        res.redirect('/books/loaned?ret=success');
+                    })
+                });
+            }
+        });
+    } catch {
+        res.redirect('/books/loaned?ret=failure');
+    }
+
 }
 
 var searchBooks = (req, res) => {
     // TODO: extract search details
     // query book model on these details
     // render page with the above details
+    let title_query = req.body.title.trim(),
```

Comment:
The search feature does not work properly. It searches only for the first word of the argument.